pyanaconda/modules/storage/zfcp/discover.py

```diff
@@ -51,18 +51,26 @@ def _check_input(self):
         if not DASD_DEVICE_NUMBER.match(self._device_number):
             raise StorageDiscoveryError("Incorrect format of the given device number.")
 
-        if not ZFCP_WWPN_NUMBER.match(self._wwpn):
+        if self._wwpn and not ZFCP_WWPN_NUMBER.match(self._wwpn):
             raise StorageDiscoveryError("Incorrect format of the given WWPN number.")
 
-        if not ZFCP_LUN_NUMBER.match(self._lun):
+        if self._lun and not ZFCP_LUN_NUMBER.match(self._lun):
             raise StorageDiscoveryError("Incorrect format of the given LUN number.")
 
+        if not ((self._device_number and not self._wwpn and not self._lun)
+                or (self._device_number and self._wwpn and self._lun)):
```

Comment:
I think it might be more readable if we knew what's actually being tested...
```
wwpn_lun_only_one_set = bool(self._wwpn) != bool(self._lun)
if not self._device_number or wwpn_lun_only_one_set:
   raise (...)
```

I also think this might be better at the front, since it's cheaper to compute than the regexps.

Reply:
I don't think that's better. If we really want to simplify the condition, the device number can be checked separately with a message `"The required device number is missing."` or something. Anyway, I am fine with the current solution.